opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -123,7 +125,7 @@ namespace Opm {
 
 
         const UnitSystem& m_deckUnitSystem;
-        const ParseContext& m_parseContext;
+        const ParseContext m_parseContext;
```

Comment:
> moves are simply pointer-copies that are "marked" for non-delete 

the pointers of the source object are not marked anything, they are simply "ripped out" of the source object, i.e. the pointers are usually set to NULL (which works with delete, but don't try to use the source object after this ;)).

>  But if an object doesn't point to any larger blobs of memory it's typically tiny anyway, meaning copies aren't expensive because most objects only contain a handful of tiny things (typically numbers).

true. in the case at hand I don't think it would affect performance even if `ParseContext` was an extremely complex beast like `Deck`. On the other hand, if that happens for objects that are used in tight loops, you *will* have a performance impact. (which is what I mean with "bad habits".)

> And at the end of the day we're going to copy here sooner or later, and if that is done in the constructor argument some opportunities are enabled, but in the worse case we should have no difference compared to reference-then-copy.

I agree that this object needs to be independent of the scope of the objects passed. I still think that using a const reference argument for the constructor is slightly nicer, though...